fastlane/spec/actions_specs/create_xcframework_spec.rb

```diff
@@ -118,6 +118,31 @@
         end
       end
 
+      context "when rewriting existing xcframework" do
+        before(:each) do
+          allow(File).to receive(:exist?).with('FrameworkA.framework').and_return(true)
+          allow(File).to receive(:exist?).with('FrameworkB.framework').and_return(true)
+          allow(File).to receive(:directory?).with('FrameworkA.framework').and_return(true)
+          allow(File).to receive(:directory?).with('FrameworkB.framework').and_return(true)
+          allow(File).to receive(:directory?).with('UniversalFramework.xcframework').and_return(true)
+        end
+
+        it "should fail due to the deleted xcframework" do
+          begin
+            Fastlane::FastFile.new.parse("lane :test do
+              create_xcframework(
+                frameworks: ['FrameworkA.framework', 'FrameworkB.framework'],
+                output: 'UniversalFramework.xcframework'
+              )
+            end").runner.execute(:test)
+          rescue => e
+            expect(e.to_s).to eq("No such file or directory @ apply2files - UniversalFramework.xcframework").or(eq("No such file or directory @ unlink_internal - UniversalFramework.xcframework"))
```

Comment:
Why is this being raised? It didn't supposed to delete the previous xcframework silently? Maybe I'm missing something here.

Reply:
@minuscorp @Kukurijek I add a commit to expect `FileUtils.remove_dir` to be called that removes the testing for failure here. I think this change should work 🤷‍♂️ 

I also updated some existing tests while I was in here 😛 The testing of failures now use an rspec way to catching and testing errors.